src/frontend/use-replicache.ts

```diff
@@ -1,32 +1,35 @@
 import { useEffect, useState } from "react";
-import { MutatorDefs, Replicache } from "replicache";
+import { MutatorDefs, Replicache, ReplicacheOptions } from "replicache";
 import { getPokeReceiver } from "./poke.js";
 
+export interface UseReplicacheOptions<M extends MutatorDefs>
+  extends Omit<ReplicacheOptions<M>, "licenseKey" | "name"> {
+  name?: string;
+}
+
 /**
  * Returns a Replicache instance with the given configuration.
- * If spaceID is falsey, returns null.
+ * If name is undefined, returns null.
  */
-export function useReplicache<M extends MutatorDefs>(
-  spaceID: string | null | undefined,
-  mutators: M
-) {
+export function useReplicache<M extends MutatorDefs>({
+  name,
+  ...options
```

Comment:
Sorry about the slow review. Digging myself back out from a big pile of stuff.

I think we should make `options` a dependency of `useReplicache`. This seems like more idiomatic React. Callers should either pass a constant options (e.g.., one declared at module level) so that it never changes, or more Reactier, pass one from `useMemo()` or `useRef()` depending on their needs.

Let me know if you agree. You've already done so much here I'll make the change.

Reply:
Also @grgbkr -- opinion wanted.

Reply:
@aboodman I'm good either way. I think it's just a slightly less straight forward DX to require useMemo/useRef in those cases. I'd bet on a few people spinning their wheels to figure out why things aren't working unless it's really obviously documented.